Vue 2 types missing type parameter constraints

### Version
2.7.9

### Reproduction link
[github.com](https://github.com/sandersn/vue2.7.9-bug)







### Steps to reproduce
```sh
$ npm i
$ npm i -g typescript@rc
$ tsc
```

### What is expected?
No compile error on Typescript 4.8

### What is actually happening?
Compile error on Typescript 4.8

node_modules/vue/types/options.d.ts(109,5): error TS2344: Type 'Mixin' does not satisfy the constraint 'ComponentOptionsMixin'.
node_modules/vue/types/options.d.ts(120,7): error TS2344: Type 'Mixin' does not satisfy the constraint 'ComponentOptionsMixin'.
node_modules/vue/types/options.d.ts(146,5): error TS2344: Type 'Mixin' does not satisfy the constraint 'ComponentOptionsMixin'.
node_modules/vue/types/options.d.ts(157,7): error TS2344: Type 'Mixin' does not satisfy the constraint 'ComponentOptionsMixin'.

---
found in Definitely Typed's overnight run:
https://dev.azure.com/definitelytyped/DefinitelyTyped/_build/results?buildId=137502&view=logs&j=6399d89d-3d30-5fff-3e92-07bbaf7ebe2f&t=482eae7c-85d2-5e8a-ea70-a5ab41e9bc13&l=12739

Vue 2 dependents all broke on Typescript 4.8 (RC) and 4.9 (in development).

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/types/options.d.ts
+++ b/types/options.d.ts
@@ -95,8 +95,8 @@ export type ThisTypedComponentOptionsWithArrayProps<
   Computed,
   PropNames extends string,
   SetupBindings,
-  Mixin,
-  Extends
+  Mixin extends ComponentOptionsMixin,
+  Extends extends ComponentOptionsMixin
 > = object &
   ComponentOptions<
     V,
@@ -132,8 +132,8 @@ export type ThisTypedComponentOptionsWithRecordProps<
   Computed,
   Props,
   SetupBindings,
-  Mixin,
-  Extends
+  Mixin extends ComponentOptionsMixin,
+  Extends extends ComponentOptionsMixin
 > = object &
   ComponentOptions<
     V,

